lib/shared/addon/components/form-pod-affinity-k8s/component.js

```diff
@@ -0,0 +1,205 @@
+          set(term, 'term', this.addWeight(term.term))
+
+          addTo.pushObject(term.term)
+        } else {
+          const removeFrom = get(this.podAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+          const addTo = get(this.podAffinity, 'requiredDuringSchedulingIgnoredDuringExecution')
+
+          set(this.podAffinity, 'preferredDuringSchedulingIgnoredDuringExecution', this.removeFromSpec(term.term, removeFrom))
+          set(term, 'term', this.removeWeight(term))
+
+          addTo.pushObject(term.term)
+        }
+        this.notifyPropertyChange('podAffinity')
+      }
+    },
+
+    antiChanged(term, old){
+      if (old){
+        if (term.priority === TERM_PRIORITY.REQUIRED){
+          const removeFrom = get(this.podAntiAffinity, 'requiredDuringSchedulingIgnoredDuringExecution')
+          const addTo = get(this.podAffinity, 'requiredDuringSchedulingIgnoredDuringExecution')
+
+          set(this.podAntiAffinity, 'requiredDuringSchedulingIgnoredDuringExecution', this.removeFromSpec(term.term, removeFrom))
+
+          addTo.pushObject(term.term)
+        } else {
+          const removeFrom = get(this.podAntiAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+          const addTo = get(this.podAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+
+          set(this.podAntiAffinity, 'preferredDuringSchedulingIgnoredDuringExecution', this.removeFromSpec(term.term, removeFrom))
+
+          addTo.pushObject(term.term)
+        }
+      } else {
+        if (term.priority === TERM_PRIORITY.REQUIRED){
+          const removeFrom = get(this.podAffinity, 'requiredDuringSchedulingIgnoredDuringExecution')
+          const addTo = get(this.podAntiAffinity, 'requiredDuringSchedulingIgnoredDuringExecution')
+
+          set(this.podAffinity, 'requiredDuringSchedulingIgnoredDuringExecution', this.removeFromSpec(term.term, removeFrom))
+
+          addTo.pushObject(term.term)
+        } else {
+          const removeFrom = get(this.podAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+          const addTo = get(this.podAntiAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+
+          set(this.podAffinity, 'preferredDuringSchedulingIgnoredDuringExecution', this.removeFromSpec(term.term, removeFrom))
+
+          addTo.pushObject(term.term)
+        }
+      }
+      this.notifyPropertyChange('podAffinity')
+      this.notifyPropertyChange('podAntiAffinity')
+    }
+  },
+
+  removeFromSpec: (term, array) => {
+    return array.filter((each) => each._id !== term._id)
+  },
+
+  addWeight: (term) => {
+    const out = {
+      // weight:          null,
+      podAffinityTerm: term,
+      _id:             term._id
+    }
+
+    delete out.podAffinityTerm._id
+
+    return out
+  },
+
+  removeWeight: (term) => {
+    const out = {
+      _id: term._id,
+      ...term.podAffinityTerm
+    }
+
+    return out
```

Comment:
It doesn't look like `out` is required

```suggestion
    return {
      _id: term._id,
      ...term.podAffinityTerm
    }
```